tests/sentry/api/endpoints/test_sentry_apps.py

```diff
@@ -17,531 +22,436 @@
 
-    def test_user_filter_on_unpublished(self):
-        self.login_as(user=self.user)
-        url = f"{self.url}?status=unpublished"
-        response = self.client.get(url, format="json")
-
-        assert response.status_code == 200
-        assert {
-            "name": self.unpublished_app.name,
-            "author": self.unpublished_app.author,
-            "slug": self.unpublished_app.slug,
-            "scopes": [],
-            "events": [],
-            "status": self.unpublished_app.get_status_display(),
-            "uuid": self.unpublished_app.uuid,
-            "webhookUrl": self.unpublished_app.webhook_url,
-            "redirectUrl": self.unpublished_app.redirect_url,
-            "isAlertable": self.unpublished_app.is_alertable,
-            "verifyInstall": self.unpublished_app.verify_install,
-            "clientId": self.unpublished_app.application.client_id,
-            "clientSecret": self.unpublished_app.application.client_secret,
-            "overview": self.unpublished_app.overview,
-            "allowedOrigins": [],
-            "schema": {},
-            "owner": {"id": self.org.id, "slug": self.org.slug},
-            "featureData": [
-                {
-                    "featureId": 0,
-                    "featureGate": "integrations-api",
-                    "description": "Testin can **utilize the Sentry API** to pull data or update resources in Sentry (with permissions granted, of course).",
-                }
-            ],
-            "popularity": self.default_popularity,
-            "avatars": [],
-        } in json.loads(response.content)
 
+class GetSentryAppsTest(SentryAppsTest):
+    def setUp(self):
+        super().setUp()
+        self.set_up_apps()
+        self.login_as(self.user)
+
+    def test_users_see_published_apps(self):
+        response = self.get_success_response()
+        self.assert_response_has_serialized_sentry_app(
+            response=response,
+            sentry_app=self.published_app,
+            organization=self.organization,
+            has_features=True,
+        )
+
+    def test_users_filter_on_internal_apps(self):
+        self.set_up_internal_app()
+        response = self.get_success_response(qs_params={"status": "internal"})
+        self.assert_response_has_serialized_sentry_app(
+            response=response,
+            sentry_app=self.internal_app,
+            organization=self.internal_organization,
+        )
         response_uuids = {o["uuid"] for o in response.data}
         assert self.published_app.uuid not in response_uuids
+        assert self.unpublished_app.uuid not in response_uuids
         assert self.unowned_unpublished_app.uuid not in response_uuids
 
-    def test_user_filter_on_published(self):
-        self.login_as(user=self.user)
-        url = f"{self.url}?status=published"
-        response = self.client.get(url, format="json")
+    def test_user_filter_on_unpublished(self):
```

Comment:
nit: I think adding `_apps` would match the test titles above this. Or you can change those ones, up to you.